model/storage/RdsStorage.php

```diff
@@ -236,9 +237,55 @@ protected function saveData(Revision $revision, array $data)
      * @return array
      */
     public function getResourcesUriByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL)
+    {
+        $result = $this->getSelectedResourcesDataByQuery(
+            [self::DATA_RESOURCE],
+            $query,
+            $options,
+            $predicate,
+        );
+
+        $resourcesUri = [];
+        while ($statement = $result->fetch()) {
+            $resourcesUri[] = $statement[self::DATA_RESOURCE];
+        }
+        return $resourcesUri;
+    }
+
+    public function getResourcesDataByQuery(string $query, array $options = [], string $predicate = OntologyRdfs::RDFS_LABEL): array
+    {
+        $result = $this->getSelectedResourcesDataByQuery(
+            [self::DATA_RESOURCE, self::DATA_OBJECT],
+            $query,
+            $options,
+            $predicate,
+        );
+
+        $resourcesData= [];
+        /** @var Revision $statement */
+        while ($statement = $result->fetch()) {
+            $resourcesData[] = [
+                'id' => $statement[self::DATA_RESOURCE],
+                'label' => $statement[self::DATA_OBJECT],
+            ];
+        }
+        return $resourcesData;
+    }
+
+    /**
+     * @param string[] $selectedFields
+     */
+    public function getSelectedResourcesDataByQuery(
```

Comment:
Any reason to be public? 
```suggestion
    private function getSelectedResourcesDataByQuery(
```